feat: refactor and add protocol support to `stats/base/stdevpn`

amishhaa opened the pull request:
Resolves #5683

## Description

This pull request adds add accessor arrays support and refactors stats/base/stdevpn.


## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5683 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

amishhaa committed:
add support for accessor arrays and refactor stats/base/stdevpn

amishhaa committed:
refactor and add protocol support to stats/base/stdevpn

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/stdevpn/README.md:
Carry out  all changes as per [Ref](https://github.com/stdlib-js/stdlib/pull/5908/files)

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/stdevpn/docs/repl.txt:
Carry out all changes as per [Ref](https://github.com/stdlib-js/stdlib/pull/5908/files)

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/stdevpn/docs/types/test.ts:
```diff
@@ -101,6 +103,7 @@ import stdevpn = require( './index' );
 	const x = new Float64Array( 10 );
 
 	stdevpn.ndarray( x.length, 1, x, 1, 0 ); // $ExpectType number
+	stdevpn( x.length, new AccessorArray( x ), 1 ); // $ExpectType number
```
```suggestion
	stdevpn.ndarray( x.length, 1, new AccessorArray( x ), 1, 0 ); // $ExpectType number
```

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/stdevpn/docs/types/test.ts:
```diff
@@ -26,6 +27,7 @@ import stdevpn = require( './index' );
 	const x = new Float64Array( 10 );
 
 	stdevpn( x.length, 1, x, 1 ); // $ExpectType number
+	stdevpn( x.length, new AccessorArray( x ), 1 ); // $ExpectType number
```
```suggestion
	stdevpn( x.length, 1, new AccessorArray( x ), 1 ); // $ExpectType number
```

gururaj1512 commented:
/stdlib merge

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/stdevpn/test/test.ndarray.js:
Independent accessor array test cases have to be added

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/stdevpn/test/test.stdevpn.js:
Same.

gururaj1512 committed:
fix: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
refactor: remove unused accessors module from stdevpn

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
docs: improve clarity in standard deviation documentation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
style: format benchmark files for consistency

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
docs: update parameter name in stdevpn documentation for clarity

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
test: add accessor array tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
Merge branch 'develop' into issue5683

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

gururaj1512 reviewed (dismissed):
LGTM Thanks...!

gururaj1512 commented:
Needs https://github.com/stdlib-js/stdlib/issues/5689 to be resolved to be merged

gururaj1512 committed:
Update lib/node_modules/@stdlib/stats/base/stdevpn/docs/types/test.ts

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

gururaj1512 committed:
Update lib/node_modules/@stdlib/stats/base/stdevpn/docs/types/test.ts

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into issue5683

kgryte commented on lib/node_modules/@stdlib/stats/base/stdevpn/docs/repl.txt:
```diff
@@ -46,27 +46,24 @@
 
     While typed array views mandate a view offset based on the underlying
-    buffer, the `offset` parameter supports indexing semantics based on a
+    buffer, the offsetX parameter supports indexing semantics based on a
```
```suggestion
    buffer, the offset parameter supports indexing semantics based on a
```

kgryte committed:
docs: update copy

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/stdevpn/docs/repl.txt:
```diff
@@ -109,8 +106,7 @@
 
     // Using offset parameter:
     > var x = [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ];
```
```suggestion
    > x = [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ];
```

kgryte committed:
docs: remove duplicate declaration

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
